### PR TITLE
Add hook executor helper package

### PR DESCRIPTION
**Pull request by hossainemruz**
Signed-off-by: Emruz Hossain <emruz@appscode.com>

### CODE REVIEW

File: pkg/invoker/restoresession.go [hunk 1]
```diff
@@ -312,28 +316,116 @@ func (inv *RestoreSessionInvoker) UpdateStatus(status RestoreInvokerStatus) erro
...
+		Target:    target,
+		Invoker: core.TypedLocalObjectReference{
+			APIGroup: pointer.StringP(v1beta1.SchemeGroupVersion.Group),
+			Kind:     v1beta1.ResourceKindRestoreSession,
+			Name:     inv.restoreSession.Name,
+		},
+	}
+	restoreSession, err := inv.stashClient.StashV1beta1().RestoreSessions(session.Namespace).Get(context.TODO(), session.Name, metav1.GetOptions{})
+	if err != nil {
+		summary.Status.Phase = string(v1beta1.RestoreFailed)
```
Comment (piyush1146115) [inline pkg/invoker/restoresession.go:332]: Can we return anything like `unknown` here?

File: pkg/conditions/restore_invoker.go [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+		Message:            msg,
+		LastTransitionTime: metav1.Now(),
+	})
+}
+
+func SetRestoreCompletedConditionToFalse(inv invoker.RestoreInvoker, tref *v1beta1.TargetRef, msg string) error {
+	return inv.SetCondition(tref, kmapi.Condition{
+		Type:               v1beta1.RestoreCompleted,
+		Status:             core.ConditionFalse,
+		Reason:             "PostRestoreTasksNotExecuted",
```
[the remark this answers is not in the document]
Comment (hossainemruz) [reply pkg/conditions/restore_invoker.go:131]: Done

File: pkg/invoker/backup_invoker.go [hunk 3]
```diff
@@ -191,7 +192,77 @@ func calculateBackupInvokerPhase(driver v1beta1.Snapshotter, conditions []kmapi.
...
+func getTargetBackupSummary(stashClient cs.Interface, target v1beta1.TargetRef, session kmapi.ObjectReference) *v1beta1.Summary {
+	summary := &v1beta1.Summary{
+		Name:      session.Name,
+		Namespace: session.Namespace,
+		Target:    target,
+	}
+
+	backupSession, err := stashClient.StashV1beta1().BackupSessions(session.Namespace).Get(context.TODO(), session.Name, metav1.GetOptions{})
+	if err != nil {
+		summary.Status.Phase = string(v1beta1.BackupSessionFailed)
```
[the remark this answers is not in the document]
Comment (hossainemruz) [reply pkg/invoker/backup_invoker.go:209]: done

File: pkg/invoker/restoresession.go (hunk 1 above)
Comment (hossainemruz) [reply pkg/invoker/restoresession.go:332]: done

**Review by piyush1146115 — APPROVED** [review]

**Pull request merged**